### PR TITLE
Charging profiles issues

### PR DESCRIPTION
**Pull request by OliveGerste**
@LucasBrazi06 
Hi Serge, I fixed the cp authorization issues by adding the _siteId_ as argument to the authorization method.
The cp loading issue was fixed by adding the _siteId_ to the db lookup like it is done e.g. in charging stations.
I tried to test everything with basic and admin user and did not find any error.
Could you double check?

### CODE REVIEW

**Comment by LucasBrazi06** [comment]
Have you taken into account the fact that the organization may not be activated?

File: src/storage/mongodb/ChargingStationStorage.ts [hunk 1]
```diff
@@ -762,6 +762,17 @@ export default class ChargingStationStorage {
       // Convert
       DatabaseUtils.pushConvertObjectIDToString(aggregation, 'chargingStation.siteAreaID');
       DatabaseUtils.pushConvertObjectIDToString(aggregation, 'chargingStation.siteArea.siteID');
+      // Check Site ID
+      if (!Utils.isEmptyArray(params.siteIDs)) {
```
Comment (LucasBrazi06) [inline src/storage/mongodb/ChargingStationStorage.ts:766]: Filter has always to be added at the beginning

File: src/server/rest/service/security/ChargingStationSecurity.ts [hunk 2]
```diff
@@ -209,7 +209,7 @@ export default class ChargingStationSecurity {
       return null;
     }
     // Check auth
-    if (Authorizations.canReadChargingProfile(loggedUser)) {
+    if (Authorizations.canReadChargingProfile(loggedUser, chargingProfile.chargingStation.siteArea.siteID)) {
```
Comment (LucasBrazi06) [inline src/server/rest/service/security/ChargingStationSecurity.ts:212]: Handle the Site Area = null

**Comment by OliveGerste** [comment]
> Have you taken into account the fact that the organization may not be activated?

Indeed there was a problem with that. (empty objects in the response)
I implemented a check in the cp filter method which removes empty objects in the response.

**Comment by OliveGerste** [comment]
> Check my comments.
> You also wrote that you tested with Basic and Admin, did you mean Site Admin?

I tested admin role and 
basic role combined with site admin.

File: src/storage/mongodb/ChargingStationStorage.ts [hunk 3]
```diff
@@ -707,6 +707,33 @@ export default class ChargingStationStorage {
...
+      // Convert
+      DatabaseUtils.pushConvertObjectIDToString(aggregation, 'chargingStation.siteAreaID');
+      DatabaseUtils.pushConvertObjectIDToString(aggregation, 'chargingStation.siteArea.siteID');
+      // Check Site ID
+      if (!Utils.isEmptyArray(params.siteIDs)) {
+        // Build filter
+        aggregation.push({
+          $match: {
+            'chargingStation.siteArea.siteID': {
+              $in: params.siteIDs.map((id) => id)
```
Comment (LucasBrazi06) [inline src/storage/mongodb/ChargingStationStorage.ts:730]: How did you manage to test when the Side ID is an ObjectID and the conversion here is missing?
I did the change but you have to test it again:
image
Let me know

File: src/storage/mongodb/ChargingStationStorage.ts [hunk 4]
```diff
@@ -707,6 +707,33 @@ export default class ChargingStationStorage {
...
+        tenantID, aggregation, localField: 'chargingStationID', foreignField: '_id',
+        asField: 'chargingStation', oneToOneCardinality: true, oneToOneCardinalityNotNull: false
+      });
+      // Site Areas
+      DatabaseUtils.pushSiteAreaLookupInAggregation({
+        tenantID, aggregation, localField: 'chargingStation.siteAreaID', foreignField: '_id',
+        asField: 'chargingStation.siteArea', oneToOneCardinality: true, oneToOneCardinalityNotNull: false
+      });
+      // Convert
+      DatabaseUtils.pushConvertObjectIDToString(aggregation, 'chargingStation.siteAreaID');
```
Comment (LucasBrazi06) [inline src/storage/mongodb/ChargingStationStorage.ts:722]: This you can let it at the end.
If you don't get the meaning of these lines your won't know where to put them.
I moved them.

**Pull request merged**